### ISSUE TITLE
Backspace while digitizing a circular arc causes QGIS to stop responding

### ISSUE THREAD

**Issue by TMcManus2018** [qgis/QGIS#35674]
## User Feedback

On a clean profile, created a polygon layer shapefile. Digitized a single line segment and then a sequence of circlular arcs. Tried to back out last point by hitting backspace. QGIS sstopped responding. 

## Report Details

**Crash ID**: [e4162199c3be8e973894328e593f88979dd1bfbf]('https://github.com/qgis/QGIS/search?q=e4162199c3be8e973894328e593f88979dd1bfbf&type=Issues')

**Stack Trace**
<pre>
QListData::begin :
QgsRubberBand::numberOfVertices qgsrubberband.cpp:608
QgsMapToolCapture::undo qgsmaptoolcapture.cpp:583
QgsMapToolCapture::keyPressEvent qgsmaptoolcapture.cpp:625
QgsMapToolAddCircularString::keyPressEvent qgsmaptooladdcircularstring.cpp:93
QgsMapCanvas::keyPressEvent qgsmapcanvas.cpp:1376
QWidget::event :
QFrame::event :
QAbstractScrollArea::event :
QgsMapCanvas::event qgsmapcanvas.cpp:2359
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify qgsapplication.cpp:418
QCoreApplication::notifyInternal2 :
QSizePolicy::QSizePolicy :
QSizePolicy::QSizePolicy :
QApplicationPrivate::notify_helper :
QApplication::notify :
QgsApplication::notify qgsapplication.cpp:418
QCoreApplication::notifyInternal2 :
QGuiApplicationPrivate::processKeyEvent :
QWindowSystemInterface::sendWindowSystemEvents :
QEventDispatcherWin32::processEvents :
UserCallWinProcCheckWow :
DispatchMessageWorker :
QEventDispatcherWin32::processEvents :
qt_plugin_query_metadata :
QEventLoop::exec :
QCoreApplication::exec :
main main.cpp:1609
BaseThreadInitThunk :
RtlUserThreadStart :
</pre>

**QGIS Info**
QGIS Version: 3.12.1-Bucure?ti
QGIS code revision: 121cc00ff0
Compiled against Qt: 5.11.2
Running against Qt: 5.11.2
Compiled against GDAL: 3.0.4
Running against GDAL: 3.0.4

**System Info**
CPU Type: x86_64
Kernel Type: winnt
Kernel Version: 10.0.18362
- [ ] Create a light and self-contained sample dataset and project file which demonstrates the issue
-->

**Describe the bug**
QGIS stops responding in middle of digitizing a polygon when attempting to backup one point by hitting the backspace key.

**How to Reproduce**

<!-- Steps, sample datasets and qgis project file to reproduce the behavior. Screencasts or screenshots welcome

1. In a clean profile
2. Create a layer with polygon features
3. digitize a line segment
4. switch to circular arc digitizing
5. digitize a number of circular arcs
6. hit backspace key (should remove latest digitized point)
7. QGIS stops responding

Note: not sure whether this happens on even numbered or odd numbered points.

**QGIS and OS versions**

See above crash report.

**Additional context**

![QGIS arc rubberbanding error](https://user-images.githubusercontent.com/35205784/78835188-9505bc80-79a4-11ea-8069-3078b11192d1.PNG)

**Comment by gioman** [comment]
>  created a polygon layer shapefile. Digitized a single line segment and then a sequence of circlular arcs.

@TMcManus2018 this is not clear to me. You started drawing a polygon (with the "add polygon" tool?) and after drawing the first segment you changed to a different digitizing tool?

A csreencast would help understand, thanks.

**Comment by TMcManus2018** [comment]
Here is a screen cast showing the problem. Keystrokes are not obvious from this. After the third arc was entered I hit Backspace. At this point the app hung (not responding). I then closed the app with the resulting QGIS error reported.
[16.04.2020_14.01.56_REC.mp4.zip](https://github.com/qgis/QGIS/files/4489750/16.04.2020_14.01.56_REC.mp4.zip)

**Comment by gioman** [comment]
@TMcManus2018 I can't replicate. The backspace key does not work for circular string, and that may be a bug or not.

When you create a shapefile you should not just select its name, you should also click on the "..." button and choose from the beginning where to save it.

**Comment by TMcManus2018** [comment]
I reproduced the error again, this time with a brand new clean profile. No additional plugins enabled. Enabled the Shape Digitizing toolbar. Using the Add Circular String tool when error occurs. I reproduce the error every time. There must (should) be some way to correct digitizing errors by "backing up" somehow, either with backspace or delete or something. Escape seems to work but that throws away the entire feature digitized to that point. No a problem for small features but annoying, just the same.

Tried it by selecting the file name with the ... just to be sure this was not part of the problem. Error still occurs.

Must be some sort of a bug as no UI action should cause the application to hang and crash. I will avoid this by carefully controlling digitizing to never try to back up. On an error, I'll just cancel out with escape and start again. Have gotten really careful about saving frequently and before i try anything new to minimize loss of work in the event of a crash.

This is not a high priority item. I'm just doing my bit to report a possible problem. It seems like it's the least I can do. Thanks for your review and efforts so far on this. If there is anything else you'd like me to do to help identify this, I'll be happy to do whatever I can.
